Allow unconfined and confined users bind ICMP sockets to generic nodes in node_t domain

Koncpa opened the pull request:
Commit 1 - Allow unconfined_t to node_bind icmp_sockets in node_t domain 
When uncofined user run ping or traceroute, this process get label unconfined_t.
Allow to ping or traceroute, which run as unconfined_t, to node_bind icmp_sockets in node_t domain.

Commit 2 - Create macro corenet_icmp_bind_generic_node()
This macro allowing bind ICMP sockets to generic nodes in node_t domain.

Commit 3 -  Allow traceroute_t and ping_t to bind generic nodes.
Use newly created macro corenet_icmp_bind_generic_node() for ping_t and traceroute_t.
This macro allowing bind generic nodes in node_t domain.


Bugzila: https://bugzilla.redhat.com/show_bug.cgi?id=1848929#c0

Koncpa committed:
Allow unconfined_t to node_bind icmp_sockets in node_t domain

When uncofined user run ping or traceroute, this process get label unconfined_t.
Allow to ping or traceroute, which run as unconfined_t, to node_bind icmp_sockets in node_t domain.

Bugzila: https://bugzilla.redhat.com/show_bug.cgi?id=1848929#c0

zpytela commented:
@koncpa, could you make ping/traceroute working for common confined users, too?

Also the commit message may fancy some polishing and `/note_bind/s/t/d/`.

Koncpa committed:
Create macro corenet_icmp_bind_generic_node()

This macro allowing bind ICMP sockets to generic nodes in node_t domain.

Koncpa committed:
Allow traceroute_t and ping_t to bind generic nodes.

Use newly created macro corenet_icmp_bind_generic_node() for ping_t and traceroute_t.
This macro allowing bind generic nodes in node_t domain.

Koncpa commented:
@zpytela, I add fix for confined users too and also I change a commit message. I hope, that now will be OK.

zpytela commented:
@koncpa, have you tested the PR with all selinux users?
Looks reasonably better; for related stuff, I'd prefer 1 squashed commit, but np with that.
In the commit messages, try to stick to these rules if possible:
https://chris.beams.io/posts/git-commit/#seven-rules

Koncpa commented:
For staff_u, sysadm_u and user_u it works. But for xguest_u, it's not working. And for guest_u is forbid network by default. Should I allow for xguest_u @zpytela ?

zpytela commented:
@Koncpa, I don't think so, see
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/selinux_users_and_administrators_guide/sect-security-enhanced_linux-targeted_policy-confined_and_unconfined_users

LGTM.
